gadget-container/entrypoint.sh

```diff
@@ -86,6 +97,14 @@ if [ "$RUNC_HOOK_MODE" = "flatcar_edge" ] ||
     echo "Installing $i..."
     cp /bin/$i /host/opt/bin/
   done
+
+  if [ "$RUNC_HOOK_MODE" = "crio" ] ; then
+    echo "Installing OCI hooks configuration in /etc/containers/oci/hooks.d/"
+    mkdir -p /host/etc/containers/oci/hooks.d/
+    cp /opt/crio-hooks/gadget-prestart.json /host/etc/containers/oci/hooks.d/gadget-prestart.json
```

Comment:
Is this path always the same?
According to [1] it's configured by `hooks_dir` in libpodman.conf. If the configuration is missing, it default is `/usr/share/containers/oci/hooks.d` according to [2].

I tried to look for `hooks_dir` in `usr/share/containers/libpod.conf` in the OpenShift playground but it appears not to be defined, do you know why it is taking `etc/containers/oci/hooks.d/` then?

Do you think we should try to parse libpod.conf to look for the configured path?

[1] https://github.com/containers/libpod/blob/master/docs/source/markdown/libpod.conf.5.md#options
[2] https://github.com/containers/libpod/blob/master/pkg/hooks/docs/oci-hooks.5.md#directories

Reply:
I am not sure how it works but I see both in the sources:

https://github.com/containers/common/blob/6d4044e24e6860e92190c29bad9468e923c99d35/pkg/config/default.go#L64-L65
```
        DefaultHooksDirs = []string{"/usr/share/containers/oci/hooks.d"}
```

https://github.com/containers/libpod/blob/1077d2d0b72d027a01080c0d3ebdb7d99a969661/pkg/hooks/hooks.go#L25-L26
```
	// OverrideDir is the directory for hook configuration files overriding the default entries.
	OverrideDir = "/etc/containers/oci/hooks.d"
```

I think we can only write in `/etc` because `/usr` is read-only.

I am not sure about parsing the configuration...

Reply:
> I am not sure about parsing the configuration...

The configuration `libpod.conf` can be in different locations too:

https://github.com/containers/common/blob/6d4044e24e6860e92190c29bad9468e923c99d35/pkg/config/libpodConfig.go#L19-L33

It seems too complex to check all of that. For now I would just drop the file in `/etc/...` and if a Kubernetes installation based on cri-o requires a different path, then we'll find a solution at that time.

cc @vbatts because you had thoughts on that.

Reply:
I agree, it seems too complex to check all those configurations.

According to https://github.com/containers/libpod/blob/6668b131eb02797c24d96cc62e1a78b8202ec04d/libpod/container_internal.go#L1741, it reads those two folders only if `hooks_dir` is not set.
Maybe the best we can do is to drop the files in `/etc/` as you suggested and add a documentation telling people to be sure that file is considered in the hooks path.

Reply:
I added a small note to Documentation/install.md about it.